### PR TITLE
Merge v4.2.2 into next

### PR DESCRIPTION
**Pull request by ZedThree**
The following files conflicted, so I would be grateful if people could give them the once over:

-	include/bout/invert/laplacexy.hxx
-	include/bout/invert/laplacexz.hxx
-	include/bout/mesh.hxx
-	include/bout/sys/expressionparser.hxx
-	include/boutexception.hxx
-	include/datafile.hxx
-	include/field.hxx
-	include/output.hxx
-	include/vecops.hxx
-	src/bout++.cxx
-	src/field/fieldgenerators.cxx
-	src/field/fieldgenerators.hxx
-	src/mesh/fv_ops.cxx
-	src/mesh/impls/bout/boutmesh.cxx
-	src/mesh/interpolation.cxx
-	src/mesh/mesh.cxx
-	src/mesh/parallel/fci.cxx
-	src/sys/expressionparser.cxx
-	tests/integrated/test-vec/runtest

### CODE REVIEW

**Comment by dschwoerer** [comment]
Check-list to make sure all files have been checked
- [ ]    include/bout/invert/laplacexy.hxx
- [ ]    include/bout/invert/laplacexz.hxx
- [ ]    include/bout/mesh.hxx
- [ ]    include/bout/sys/expressionparser.hxx
- [ ]    include/boutexception.hxx
- [ ]    include/datafile.hxx
- [ ]    include/field.hxx
- [ ]    include/output.hxx
- [ ]    include/vecops.hxx
- [ ]    src/bout++.cxx
- [ ]    src/field/fieldgenerators.cxx
- [ ]    src/field/fieldgenerators.hxx
- [ ]    src/mesh/fv_ops.cxx
- [ ]    src/mesh/impls/bout/boutmesh.cxx
- [ ]    src/mesh/interpolation.cxx
- [ ]    src/mesh/mesh.cxx
- [ ]    src/mesh/parallel/fci.cxx
- [ ]    src/sys/expressionparser.cxx
- [ ]    tests/integrated/test-vec/runtest

File: include/vecops.hxx [hunk 1]
```diff
@@ -81,7 +81,7 @@ const Field2D Div(const Vector2D& v, CELL_LOC outloc = CELL_DEFAULT,
 const Field3D Div(const Vector3D& v, CELL_LOC outloc = CELL_DEFAULT,
                   const std::string& method = "DEFAULT");
 
-const Field2D Div(const Vector2D &v, const Field2D &f);
+const Field2D Div(const Vector2D& v, const Field2D& f, CELL_LOC outloc = CELL_DEFAULT);
```
Comment (johnomotani) [inline include/vecops.hxx]: Think this should have a `const std::string& method = "DEFAULT"` argument too
https://github.com/boutproject/BOUT-dev/blob/eeed651ac6ce00694d6debfe3b28ed00245bade0/src/field/vecops.cxx#L182-L183

Comment (ZedThree) [reply include/vecops.hxx]: Thanks, fixed!

**Comment by dschwoerer** [comment]
It seems you dropped the `"%s",` workaround from https://github.com/boutproject/BOUT-dev/blob/next/src/mesh/interpolation.cxx#L32 (and L33)

**Comment by ZedThree** [comment]
> It seems you dropped the `"%s",` workaround from https://github.com/boutproject/BOUT-dev/blob/next/src/mesh/interpolation.cxx#L32 (and L33)

Thanks, fixed!

File: src/mesh/parallel/fci.cxx [hunk 2]
```diff
@@ -120,38 +120,44 @@ FCIMap::FCIMap(Mesh& mesh, int offset_, BoundaryRegionPar* boundary, bool zperio
...
+    interp_corner->calcWeights(xt_prime_corner, zt_prime_corner);
+  }
 
-  int ncz = map_mesh.LocalNz;
+  {
+    TRACE("FCImap: calculating weights");
+    interp->calcWeights(xt_prime, zt_prime);
+  }
+  
+  int ncz = mesh.LocalNz;
```
Comment (ZedThree) [inline src/mesh/parallel/fci.cxx]: `mesh` -> `map_mesh`. Compiles and works fine because `map_mesh == mesh`, but best to be consistent!

**Comment by ZedThree** [comment]
Ok, I'm pretty happy for this to go in now. Would someone like to do the formalities? :slightly_smiling_face:

**Comment by bendudson** [comment]
Where is the emoji for a bottle smash? :champagne:

**Pull request merged**